src/kitovu/sync/filecache.py

```diff
@@ -0,0 +1,70 @@
+"""FileCache keeps track of the state of the files, remotely and locally. It exists to determine if the local file
+has been changed between two synchronisation processes and allows for a conflict handling accordingly."""
+
+from enum import Enum
+import json
+import pathlib
+import typing
+
+from kitovu.sync import syncplugin
+from kitovu import utils
+
+# sync-process:
+# every time we sync something, the FileCache gets read, upon which kitovu decides what needs to be downloaded.
+# The files get downloaded, for each file, the FileCache gets updated for the files that were synchronised.
+
+
+class Filestate(Enum):
+    """Used to discern in which places files have changed"""
+    LOCAL = 1
+    REMOTE = 2
+    BOTH = 3
+    NONE = 4
+
+
+class File:
+    def __init__(self, local_digest_at_synctime: str,
+                 relative_path_with_filename: pathlib.PurePath,
+                 plugin: syncplugin.AbstractSyncPlugin) -> None:
+        file_properties = {"path": relative_path_with_filename,
+                           "plugin": plugin,
+                           "digests": {
+                                "cached digest": local_digest_at_synctime,
+                                }
+                           }
+
+
+class FileCache:
+    FILECACHE: str = "filecache.json"
+
+    def write_file_to_filecache(self, file: File):
```

Comment:
You seem to try reading/writing the JSON with every operation - that's more complexity than needed, and also won't work out (you can't just dump a file to JSON and append that to the file).

Instead, I imagine things to work something like this:

- Read the entire file cache from disk once (with `json.load`) with a method like `FileCache.read()` (shouldn't need any arguments)
- Interact with the `FileCache` by adding files to it, or comparing them, etc.
- Write it to disk (with `json.dump`) once everything is finished with a method like `FileCache.write()` (or `.store()`) - again, probably without any arguments.